Binary files gpointing-device-settings/modules/gnome-settings-daemon-plugins/.gsd-pointing-device-plugin.c.swp and gpointing-device-settings-build/modules/gnome-settings-daemon-plugins/.gsd-pointing-device-plugin.c.swp differ
diff -Nru gpointing-device-settings/modules/gnome-settings-daemon-plugins/gsd-pointing-device-plugin.c gpointing-device-settings-build/modules/gnome-settings-daemon-plugins/gsd-pointing-device-plugin.c
--- gpointing-device-settings/modules/gnome-settings-daemon-plugins/gsd-pointing-device-plugin.c	2012-01-21 18:46:24.770666951 -0500
+++ gpointing-device-settings-build/modules/gnome-settings-daemon-plugins/gsd-pointing-device-plugin.c	2012-01-21 19:08:25.383684958 -0500
@@ -124,7 +124,7 @@
 
     gint op_code, event, error;
 
-    if (!XQueryExtension(GDK_DISPLAY(),
+    if (!XQueryExtension(gdk_x11_get_default_xdisplay(),
                          "XInputExtension",
                          &op_code,
                          &event,
@@ -212,6 +212,11 @@
     plugin_class->activate = activate;
     plugin_class->deactivate = deactivate;
 }
+
+static void
+gsd_pointing_device_plugin_class_finalize(GsdPointingDevicePluginClass *klass)
+{
+}
 /*
 vi:ts=4:nowrap:ai:expandtab:sw=4
 */
diff -Nru gpointing-device-settings/modules/gnome-settings-daemon-plugins/gsd-touchpad-manager.c gpointing-device-settings-build/modules/gnome-settings-daemon-plugins/gsd-touchpad-manager.c
--- gpointing-device-settings/modules/gnome-settings-daemon-plugins/gsd-touchpad-manager.c	2012-01-21 18:46:24.770666951 -0500
+++ gpointing-device-settings-build/modules/gnome-settings-daemon-plugins/gsd-touchpad-manager.c	2012-01-21 18:52:05.229671597 -0500
@@ -348,7 +348,7 @@
 
     gint op_code, event, error;
 
-    if (!XQueryExtension(GDK_DISPLAY(),
+    if (!XQueryExtension(gdk_x11_get_default_xdisplay(),
                          "XInputExtension",
                          &op_code,
                          &event,
diff -Nru gpointing-device-settings/src/gpds-xinput-pointer-info.c gpointing-device-settings-build/src/gpds-xinput-pointer-info.c
--- gpointing-device-settings/src/gpds-xinput-pointer-info.c	2012-01-21 18:46:24.775666951 -0500
+++ gpointing-device-settings-build/src/gpds-xinput-pointer-info.c	2012-01-21 18:54:37.300673676 -0500
@@ -55,7 +55,7 @@
     XDeviceInfo *device_infos;
     gint i, n_device_infos;
 
-    device_infos = XListInputDevices(GDK_DISPLAY(), &n_device_infos);
+    device_infos = XListInputDevices(gdk_x11_get_default_xdisplay(), &n_device_infos);
 
     for (i = 0; i < n_device_infos; i++) {
         GpdsXInputPointerInfo *info;
diff -Nru gpointing-device-settings/src/gpds-xinput-utils.c gpointing-device-settings-build/src/gpds-xinput-utils.c
--- gpointing-device-settings/src/gpds-xinput-utils.c	2012-01-21 18:46:24.775666951 -0500
+++ gpointing-device-settings-build/src/gpds-xinput-utils.c	2012-01-21 18:59:37.311677761 -0500
@@ -38,10 +38,12 @@
 XDeviceInfo *
 gpds_xinput_utils_get_device_info (const gchar *device_name, GError **error)
 {
+    if (!device_name)
+        goto out_gpds_xinput_utils_get_device_info;
     XDeviceInfo *device_infos;
     gint i, n_device_infos;
 
-    device_infos = XListInputDevices(GDK_DISPLAY(), &n_device_infos);
+    device_infos = XListInputDevices(gdk_x11_get_default_xdisplay(), &n_device_infos);
 
     for (i = 0; i < n_device_infos; i++) {
         if (device_infos[i].use != IsXExtensionPointer)
@@ -54,6 +56,7 @@
 
     XFreeDeviceList(device_infos);
 
+out_gpds_xinput_utils_get_device_info:
     g_set_error(error,
                 GPDS_XINPUT_UTILS_ERROR,
                 GPDS_XINPUT_UTILS_ERROR_NO_DEVICE,
@@ -68,7 +71,7 @@
     XDeviceInfo *device_infos;
     gint i, n_device_infos;
 
-    device_infos = XListInputDevices(GDK_DISPLAY(), &n_device_infos);
+    device_infos = XListInputDevices(gdk_x11_get_default_xdisplay(), &n_device_infos);
 
     for (i = 0; i < n_device_infos; i++) {
         if (device_infos[i].use != IsXExtensionPointer)
@@ -125,7 +128,7 @@
         return NULL;
 
     gdk_error_trap_push();
-    device = XOpenDevice(GDK_DISPLAY(), device_info->id);
+    device = XOpenDevice(gdk_x11_get_default_xdisplay(), device_info->id);
     gdk_error_trap_pop();
     if (!device) {
         g_set_error(error,
diff -Nru gpointing-device-settings/src/gpds-xinput.c gpointing-device-settings-build/src/gpds-xinput.c
--- gpointing-device-settings/src/gpds-xinput.c	2012-01-21 18:46:24.775666951 -0500
+++ gpointing-device-settings-build/src/gpds-xinput.c	2012-01-21 18:54:10.448673303 -0500
@@ -100,7 +100,7 @@
     g_free(priv->device_name);
 
     if (priv->device) {
-        XCloseDevice(GDK_DISPLAY(), priv->device);
+        XCloseDevice(gdk_x11_get_default_xdisplay(), priv->device);
         priv->device = NULL;
     }
 
@@ -199,7 +199,7 @@
 {
     gchar buf[64];
 
-    XGetErrorText(GDK_DISPLAY(), x_error_code, buf, sizeof(buf) - 1);
+    XGetErrorText(gdk_x11_get_default_xdisplay(), x_error_code, buf, sizeof(buf) - 1);
 
     return g_strdup(buf);
 }
@@ -273,7 +273,7 @@
     }
 
     gdk_error_trap_push();
-    XChangeDeviceProperty(GDK_DISPLAY(),
+    XChangeDeviceProperty(gdk_x11_get_default_xdisplay(),
                           device, property_atom,
                           XA_INTEGER, format_type, PropModeReplace,
                           (unsigned char*)property_data, n_properties);
@@ -383,7 +383,7 @@
     if (!device)
         return -1;
 
-    properties = XListDeviceProperties(GDK_DISPLAY(), device, &n_properties);
+    properties = XListDeviceProperties(gdk_x11_get_default_xdisplay(), device, &n_properties);
     for (i = 0; i < n_properties; i++) {
         const gchar *name;
 
@@ -427,7 +427,7 @@
         return FALSE;
 
     gdk_error_trap_push();
-    status =  XGetDeviceProperty(GDK_DISPLAY(), device, atom, 0, 1000, False,
+    status =  XGetDeviceProperty(gdk_x11_get_default_xdisplay(), device, atom, 0, 1000, False,
                                  XA_INTEGER, &actual_type, &actual_format,
                                  n_values, &bytes_after, &data);
     gdk_flush();
@@ -523,7 +523,7 @@
         *(property_data + i) = (gfloat)properties[i];
 
     gdk_error_trap_push();
-    XChangeDeviceProperty(GDK_DISPLAY(),
+    XChangeDeviceProperty(gdk_x11_get_default_xdisplay(),
                           device, property_atom,
                           float_atom, 32, PropModeReplace,
                           (unsigned char*)property_data, n_properties);
@@ -594,7 +594,7 @@
         return FALSE;
 
     gdk_error_trap_push();
-    status =  XGetDeviceProperty(GDK_DISPLAY(), device, property_atom, 0, 1000, False,
+    status =  XGetDeviceProperty(gdk_x11_get_default_xdisplay(), device, property_atom, 0, 1000, False,
                                  float_atom, &actual_type, &actual_format,
                                  n_properties, &bytes_after, &data);
     gdk_flush();
@@ -676,7 +676,7 @@
     *map = g_new0(guchar, *n_buttons);
 
     gdk_error_trap_push();
-    status =  XGetDeviceButtonMapping(GDK_DISPLAY(), device, *map, *n_buttons);
+    status =  XGetDeviceButtonMapping(gdk_x11_get_default_xdisplay(), device, *map, *n_buttons);
     gdk_flush();
 
     x_error_code = gdk_error_trap_pop();
@@ -705,7 +705,7 @@
         return FALSE;
 
     gdk_error_trap_push();
-    status =  XSetDeviceButtonMapping(GDK_DISPLAY(), device, map, n_buttons);
+    status =  XSetDeviceButtonMapping(gdk_x11_get_default_xdisplay(), device, map, n_buttons);
     gdk_flush();
 
     x_error_code = gdk_error_trap_pop();
diff -Nru gpointing-device-settings/test/test-xinput-utils.c gpointing-device-settings-build/test/test-xinput-utils.c
--- gpointing-device-settings/test/test-xinput-utils.c	2012-01-21 18:46:24.776666951 -0500
+++ gpointing-device-settings-build/test/test-xinput-utils.c	2012-01-21 18:55:41.242674547 -0500
@@ -29,7 +29,7 @@
 teardown (void)
 {
     if (device)
-        XCloseDevice(GDK_DISPLAY(), device);
+        XCloseDevice(gdk_x11_get_default_xdisplay(), device);
     g_clear_error(&error);
     g_clear_error(&expected_error);
 }
